test/http/request.cpp

```diff
@@ -292,3 +292,134 @@ TEST_CASE("The callback is called if input URL parsing fails") {
+                {"url", "http://www.google.com/"},
+            }, {}, "", [transport](Error error) {
+                REQUIRE(!error);
+                request_recv_response(transport, [](Error e, Var<Response> r) {
+                    REQUIRE(!e);
+                    REQUIRE(status_code_ok(r->status_code));
+                    REQUIRE(r->body.size() > 0);
+                    break_loop();
+                });
+            });
+        });
+    });
+}
+
+TEST_CASE("http::request_sendrecv() works as expected") {
+    loop_with_initial_event([]() {
+        request_connect({
+            {"url", "http://www.google.com/"}
+        }, [](Error error, Var<Transport> transport) {
+            REQUIRE(!error);
+            request_sendrecv(transport, {
+                {"method", "GET"},
+                {"url", "http://www.google.com/"},
+            }, {}, "", [transport](Error error, Var<Response> r) {
+                REQUIRE(!error);
+                REQUIRE(status_code_ok(r->status_code));
+                REQUIRE(r->body.size() > 0);
+                break_loop();
+            });
+        });
+    });
+}
+
+TEST_CASE("http::request_sendrecv() works for multiple requests") {
+    loop_with_initial_event([]() {
+        request_connect({
+            {"url", "http://www.google.com/"}
+        }, [](Error error, Var<Transport> transport) {
+            REQUIRE(!error);
+            request_sendrecv(transport, {
+                {"method", "GET"},
+                {"url", "http://www.google.com/"},
+            }, {}, "", [transport](Error error, Var<Response> r) {
+                REQUIRE(!error);
+                REQUIRE(status_code_ok(r->status_code));
+                REQUIRE(r->body.size() > 0);
+                request_sendrecv(transport, {
+                    {"method", "GET"},
+                    {"url", "http://www.google.com/robots.txt"},
+                }, {}, "", [transport](Error error, Var<Response> r) {
+                    REQUIRE(!error);
+                    REQUIRE(r->status_code == 200);
+                    REQUIRE(r->body.size() > 0);
+                    break_loop();
+                });
+            });
+        });
+    });
+}
+
+TEST_CASE("http::request_cycle() works as expected") {
+    loop_with_initial_event([]() {
+        request_cycle({
+            {"method", "GET"},
+            {"url", "http://www.google.com/robots.txt"}
+        }, {}, "", [](Error error, Var<Response> r) {
+            REQUIRE(!error);
+            REQUIRE(r->status_code == 200);
+            REQUIRE(r->body.size() > 0);
+            break_loop();
+        });
+    });
+}
+
+TEST_CASE("http::request_cycle() works as expected using Tor") {
+    loop_with_initial_event([]() {
+        request_cycle({
+            {"method", "GET"},
+            {"url", "httpo://www.google.com/robots.txt"},
+        }, {}, "", [](Error error, Var<Response> r) {
+            // Test disable because we don't know if Tor is running
```

Comment:
I think we should find a way to exclude this test if we don't know that tor is runnig. Maybe we can put all the tests that need tor in a test file and exclude if it is not necessary, or maybe we can define a preprocessor variable in the configure that enables this code, something like 
#ifdef TOR_TEST...